ratpack-pac4j/src/main/java/ratpack/pac4j/internal/Pac4jAuthenticator.java

```diff
@@ -81,13 +82,28 @@ public void handle(Context ctx) throws Exception {
   }
 
   public Clients createClients(Context ctx, PathBinding pathBinding) throws Exception {
+    String callback = createCallbackUrl(ctx.get(PublicAddress.class), pathBinding, path);
+
+    return new Clients(callback, getClientsFromProvider(ctx, clientsProvider));
+  }
+
+  private String createCallbackUrl(PublicAddress publicAddress, PathBinding pathBinding, String path) {
     String boundTo = pathBinding.getBoundTo();
-    PublicAddress publicAddress = ctx.get(PublicAddress.class);
-    String absoluteCallbackUrl = publicAddress.get() + boundTo + "/" + path;
+    URI address = publicAddress.get();
+
+    if(!boundTo.isEmpty()) {
+      if (!(address.toString().endsWith("/") || boundTo.startsWith("/"))) {
```

Comment:
You are correct - the two provided implementations (ConstantPublicAddress and InferringPublicAddress) both reconstruct the URI from scheme, host and port so there would be no trailing slash. However I thought that there was cause to include the check:
1. implementations of PublicAddress can be user provided 
2. the PublicAddress interface returns a URI - which contains path in its public contract. I would be happier if PublicAddress returned an interface that described only the scheme + host + path. It would not be unreasonable for users to add a path to their implementation as it is not obvious that the return type can not contain a path (especially for old skool servlet types who expect to be able to set a web application context).